### DISCUSSION TITLE
Can you share scripts to generate the OPDSynth dataset?

### DISCUSSION THREAD

**Discussion by keiohta** [3dlg-hcvc/OPD#5] [Q&A] [answered]
Hi authors, thanks again for sharing the great work and repo!

We are working on a similar project and would like to generate a dataset for our experiments following your procedure written in section C.2, but have some trouble reproducing the procedure. So, would it be possible to share the scripts to generate the dataset?

Thanks again,
Kei

**Comment by Jianghanxiao** [comment, marked as answer]
Hi, 

Thanks for asking. C.2 is about the rendering process. We use the scene-toolkit repo (https://github.com/smartscenes/sstk) to render our images. (We use some internal version for now, because the codebase is too big, we are still working on cleaning it). But  you can easily choose some other rendering library to render the images, like pyrender. 

What's the exact question you meet when rendering the images? About the viewpoint selection or the motion state setting? Can you give more information on the problem you meet

**Reply by keiohta** [reply]
Thanks for the answer. Our rendering process is 1) load URDF using SAPIEN, 2) change camera pose, and 3) save pictures and corresponding joint states. One of the problems (not actually a problem but todo) is how to change background images as you did in your paper. So, the concrete questions are
1. how can you change the background?
2. where did you get the background images?

**Reply by Jianghanxiao** [reply]
Hi,

Thanks for the questions. 

The answer to the first question is that we don't change the background during the rendering process. After rendering, we get the images, and then we add random background to the images (because we have the object masks). 

The answer to the second question is that we use the RGB images from the MatterPort3D dataset. 

These stuffs are in C.2.3 `We augment the images using RGBD backgrounds from the Matterport3D [4] dataset by randomly selecting from the ‘straight ahead’ and ‘downward tilt’ camera views`

Feel free to ask if the answers are not clear

**Reply by keiohta** [reply]
Hi @Jianghanxiao, thanks for the answers. It completely clarified my understanding and am ready to reproduce the process. Let me ask again if I have another question. Thanks again for your time!

**Reply by keiohta** [reply]
Hi @Jianghanxiao, let me ask for another detail. 

Regarding the rendering process, I want to clarify my understanding of what the `5 + 20 * num_parts` views mean. My understanding is 
- `5 + 20 * num_parts = 5 * (1 + 4 * num_parts)`
  - the first `5` means five different camera angles/positions (sampled from the Bates distribution in C.2.2)
  - the `1` means all movable parts are closed
  - the `4 * num_parts` means four random joint states for each moving part.

Is my understanding correct?

Also, does `while other parts stay at the min value of their motion range` mean only one movable part has a random state and the other parts are closed?

**Reply by Jianghanxiao** [reply]
Hi @keiohta, 

This is a great question. In SAPIEN data, there is no semantic meaning of the min value of their motion range. That's why we only demonstrate other parts stay at the min value. So there is no promise on the close or open state, because the min value is not always the close state. In other word, there can be multiple open parts in the image

**Reply by keiohta** [reply]
Hi @Jianghanxiao ,

Thanks for the answer. I understand the reason why more than one parts are sometimes open state in the image. Thanks for the clarification!
